commit bd86d7c2841a1e1bc313c31a4bce2740f1b58f4f
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2019-03-21T16:44:10-07:00 (7 months ago)
    
    MANTA-4172 zk client ping timeouts not working
    MANATEE-375 Update dev zone scripts for MANATEE-335
    MANATEE-343 manatee full rebuilds should pull from the primary where possible
    MANATEE-400 Manatee deleterious when dataset will not unmount
    MANATEE-405 snapshotter should not enter maintenance when dataset does not exist
    MANATEE-307 manatee-snapshotter failing on snapshot with dependent clones
    MANATEE-332 manatee-snapshotter fails spuriously during rebuild
    MANATEE-397 `manatee-adm history` fails with `TypeError: Cannot read property 'zoneId' of null` (again)
    MANATEE-398 manatee-adm history -j doesn't understand v1 timestamps
    MANATEE-389 deposed array is missing in stateBackfill
    MANATEE-390 check for deposed array's existence in manatee-adm history
    MANATEE-359 manatee reports primary to sync replication as async
    MANATEE-341 "manatee-adm rebuild" hangs when rebuilding a sync peer
    MANATEE-344 Rebuild should not exit if postgres hasn't started on its first attempt

diff --git a/boot/setup.sh b/boot/setup.sh
index 49b403a..e7e3aa6 100755
--- a/boot/setup.sh
+++ b/boot/setup.sh
@@ -7,7 +7,7 @@
 #
 
 #
-# Copyright (c) 2018, Joyent, Inc.
+# Copyright (c) 2019, Joyent, Inc.
 #
 
 export PS4='[\D{%FT%TZ}] ${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
@@ -32,10 +32,7 @@ touch /opt/local/.dli_license_accepted
 function sdc_manatee_setup {
     # vars used by manatee-* tools
     ZONE_UUID=$(json -f /var/tmp/metadata.json ZONE_UUID)
-    DATASET=zones/$ZONE_UUID/data/manatee
     PARENT_DATASET=zones/$ZONE_UUID/data
-    DATASET_MOUNT_DIR=/manatee/pg
-    PG_DIR=/manatee/pg/data
     PG_LOG_DIR=/var/pg
     BINDER_ADMIN_IPS=$(json -f /var/tmp/metadata.json binder_admin_ips)
 
@@ -92,13 +89,6 @@ function common_manatee_setup {
     echo "setting recordsize to 8K on manatee dataset"
     zfs set recordsize=8k "$PARENT_DATASET"
 
-    # create manatee dataset
-    echo "creating manatee dataset"
-    zfs create -o canmount=noauto $DATASET
-
-    echo "make snapdir property match the ancestor's"
-    zfs inherit -r snapdir $DATASET
-
     # create postgres group
     echo "creating postgres group (gid=907)"
     groupadd -g 907 postgres
@@ -114,23 +104,6 @@ function common_manatee_setup {
     echo "grant postgres user zfs perms"
     zfs allow -ld postgres create,destroy,diff,hold,release,rename,setuid,rollback,share,snapshot,mount,promote,send,receive,clone,mountpoint,canmount $PARENT_DATASET
 
-    # change dataset perms such that manatee can access the dataset and mount/unmount
-    mkdir -p $DATASET_MOUNT_DIR
-    chown -R postgres $DATASET_MOUNT_DIR
-    chmod 700 -R $DATASET_MOUNT_DIR
-
-    # set mountpoint
-    zfs set mountpoint=$DATASET_MOUNT_DIR $DATASET
-
-    # mount the dataset
-    zfs mount $DATASET
-
-    # make the pg data dir
-    echo "creating $PG_DIR"
-    mkdir -p $PG_DIR
-    chown postgres $PG_DIR
-    chmod 700 $PG_DIR
-
     # add pg log dir
     mkdir -p $PG_LOG_DIR
     chown -R postgres $PG_LOG_DIR
diff --git a/package.json b/package.json
index 4db2cea..7064eaf 100644
--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
     },
     "dependencies": {
         "carrier": "0.2.0",
-        "manatee": "git+https://github.com/joyent/manatee.git#6a769d97"
+        "manatee": "git+https://github.com/joyent/manatee.git#33b33433070e1c83622ca6a555b4bc38f1eec0e6"
     },
     "sdcDependencies": {
         "config-agent": ">=1.2.0"
